File: src/main/java/com/lsd/events/PageTitle.java
package com.lsd.events;

import lombok.Value;

import static java.lang.System.lineSeparator;
import static org.apache.commons.lang3.StringUtils.isBlank;

/**
 * Used to indicate that a new page should be created in a sequence diagram.
 */
@Value(staticConstructor = "titled")
public class PageTitle implements SequenceEvent {
    String keyword = "title";
    String title;

    public String toMarkup() {
        String titleValue = isBlank(title) ? "<title missing>" : " " + title;
        return lineSeparator() + keyword + titleValue + lineSeparator();
    }
}

File: src/main/kotlin/com/lsd/events/PageTitle.kt
package com.lsd.events

import java.lang.System.lineSeparator

/**
 * Used to add a title in a sequence diagram.
 */
data class PageTitle(val title: String) : SequenceEvent {
    override fun toMarkup(): String {
        val safeTitle = title.ifBlank { "<title missing>" }
        return "${lineSeparator()}title ${safeTitle}${lineSeparator()}"
    }

    companion object {
        @JvmStatic
        fun titled(title: String): PageTitle = PageTitle(title)
    }
}